### PR TITLE
Wrap bg subtract

### PR DESCRIPTION
**Pull request by hannah413**
Initial wrap of background subtraction tool.

### CODE REVIEW

**Review by luke-c-sargent — APPROVED** [review]
looks good to me!

File: tools/backsub/backsub.xml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    </outputs>
+
+    <tests>
+        <test expect_num_outputs="2">
+            <param name="root" value="test.ome.tiff" />
+            <param name="markers" value="test_markers.csv" />
+            <param name="tile_size" value="256" />
+            <output name="image_output" ftype="ome.tiff">
+                <assert_contents>
+                    <has_size value="520000" delta="10000" />
```
Comment (bgruening) [inline tools/backsub/backsub.xml:60]: Just as information for future udpates, the latest Galaxy release has some nice image assertions: https://docs.galaxyproject.org/en/latest/dev/schema.html#assertions-for-image-data

So we can now design tests that are way more stringent.

File: tools/backsub/backsub.xml [hunk 2]
```diff
@@ -0,0 +1,96 @@
+<tool id="backsub" name="Background subtraction" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="@PROFILE@">
+
+    <description>for sequential immunofluorescence images.</description>
```
Comment (bgruening) [inline tools/backsub/backsub.xml:3]:
```suggestion
    <description>for sequential immunofluorescence images</description>
```

File: tools/backsub/backsub.xml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+ 
+    <expand macro="requirements"/>
+    <expand macro="version_cmd"/>
+
+    <command detect_errors="aggressive"><![CDATA[
+
+        @CMD_BEGIN@
```
Comment (bgruening) [inline tools/backsub/backsub.xml:14]: Imho this is a bit too much macros :) Its not helping for a review and does not reduce repetition as its only used one time.

File: tools/backsub/backsub.xml [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+    <command detect_errors="aggressive"><![CDATA[
+
+        @CMD_BEGIN@ 
+
+        ##Supply image
+        --root $root
+
+        ##Supply markers
+        --markers $markers
```
Comment (bgruening) [inline tools/backsub/backsub.xml:20]:
```suggestion
        --markers '$markers'
```

File: tools/backsub/backsub.xml [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+ 
+    <expand macro="requirements"/>
+    <expand macro="version_cmd"/>
+
+    <command detect_errors="aggressive"><![CDATA[
+
+        @CMD_BEGIN@ 
+
+        ##Supply image
+        --root $root
```
Comment (bgruening) [inline tools/backsub/backsub.xml:17]:
```suggestion
        --root '$root'
```

File: tools/backsub/backsub.xml [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+        --tile-size $adv.tile_size
+        #end if
+        
+    ]]></command>
+
+    <inputs>
+        <param name="root" type="data" format="ome.tiff" optional="false" label="Image to process"/>
+        <param name="markers" type="data" format="csv" optional="false" label="Markers file mapping channels to subtract" help="Expected columns: marker_name, background, exposure, remove"/>
+        <section name="adv" title="Advanced Options" expanded="false">
+            <param name="pixel_size" type="float" optional="true" label="Pixel size in microns" help="If not supplied, finds resolution in XML metadata, otherwise defaults to 1"/>
```
Comment (bgruening) [inline tools/backsub/backsub.xml:43]: Can you maybe add a min/max value here and below?

File: tools/backsub/backsub.xml (hunk 1 above)
Comment (CameronFRWatson) [reply tools/backsub/backsub.xml:60]: Thank you very much for the review and feedback @bgruening. Good to know about the new image assertions, I'll create an issue in our main repo so we remember to update that across our tools when we update our instances and CI to the latest Galaxy version

**Pull request merged**